### PR TITLE
New fixed point decimal parser and serializer

### LINKED ISSUE #102: Use fixed point decimal conversion for the OpenFinex API
In order to have a clean conversion between decimals and block chain / substrate integers, we must use a conversion algorithm that does not involve any floating point numbers (as we do now). We need to parse the decimal directly from string.

According to OpenFinex, they use https://github.com/shopspring/decimal in Go.

Substrate has some support for this, see https://substrate.dev/rustdocs/v3.0.0/node_template_runtime/struct.Permill.html . It is not clear however how much of this can be used. Further investigation is needed.

### PR DESCRIPTION
**Pull request by murerfel**
Had to implement my own parser and serializer for base10 fixed-point decimals, all other solutions were either base2 or introduced losses when parsing from a base10 number string.

Features:
* Parses a fixed point number, e.g. "10.2834" into a `u128` integer, shifted by `UNIT` (which is `1e18`). (lossless, if number has less than 18 fraction digits)
* Serializes a `u128` integer with `UNIT` base to a string

The limitations for this parser/serializer:
* No scientific notation (e.g. `1.4e-3`)
* No negative numbers
* 18 fraction digits (digits after decimal point) 
* Maximum integer value of `340282366920938463463`

Closes #102

### CODE REVIEW

File: enclave/src/openfinex/fixed_point_number_converter.rs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+use crate::constants::UNIT;
+use crate::openfinex::openfinex_api::{OpenFinexApiError, OpenFinexApiResult};
+use crate::openfinex::openfinex_types::ResponseInteger;
+use alloc::{string::String, string::ToString, vec::Vec};
+use core::iter::Peekable;
+
+/// Provides conversion from and to a string containing a fixed point decimal with 18-digit precision
+/// to an u128 integer, scaled by UNIT (=1e18).
+pub struct FixedPointNumberConverter {}
```
Comment (murerfel) [inline enclave/src/openfinex/fixed_point_number_converter.rs:29]: This is the new converter, i.e. parser and serializer, for fixed point number strings.
Replaces the previous implementation in `number_value_conversion.rs`

File: enclave/src/openfinex/fixed_point_number_converter.rs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+    pub fn fail_to_parse_if_too_large() {
+        // number in string is larger than what can be converted to u128 (with 18 digit shift)
+        // u128 max 340_282_366_920_938_463_463,,_374_607_431_768_211_455
+        assert!(
+            FixedPointNumberConverter::parse_from_string(&format!("340282366920938463464"))
+                .is_err()
+        );
+    }
```
Comment (murerfel) [inline enclave/src/openfinex/fixed_point_number_converter.rs:177]: Overflow test, we fail to parse if the value in the string exceeds what the `u128`, shifted by 18 digits, can hold

File: enclave/src/openfinex/fixed_point_number_converter.rs [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+    pub fn successfully_parse_numbers() {
```
Comment (murerfel) [inline enclave/src/openfinex/fixed_point_number_converter.rs:179]: Below are examples of numbers that can be parsed successfully and their corresponding integer value

**Review by haerdib — APPROVED** [review]
Looks great  to me!

**Pull request merged**